tests/unit/amo/components/TestCollection.js

```diff
@@ -886,6 +947,43 @@ describe(__filename, () => {
     }));
   });
 
+  it('dispatches deleteCollection when onDelete is called', () => {
+    const authorUserId = 11;
+    const slug = 'some-slug';
+    const username = 'some-username';
+    const { store } = dispatchSignInActions({ userId: authorUserId });
+
+    store.dispatch(loadCurrentCollection({
+      addons: createFakeCollectionAddons(),
+      detail: createFakeCollectionDetail({
+        authorId: authorUserId,
+        authorUsername: username,
+        slug,
+      }),
+    }));
+
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+    const preventDefaultSpy = sinon.spy();
+    const errorHandler = createStubErrorHandler();
+
+    const wrapper = renderComponent({ errorHandler, store });
+
+    dispatchSpy.reset();
+
+    // This emulates a user clicking the delete button and confirming.
+    wrapper.instance().onDelete(
```

Comment:
In my opinion, we should not use `instance()` that much, and especially not here. I'd rather find the `ConfirmButton` and use `.prop` to get the handler:

```js
const onDelete = wrapper.find(ConfirmButton).prop('onConfirm');
onDelete(createFakeEvent({ ... }));
```

I consider the `onDelete` method "internal" and therefore "private". If we rename this method for some reason, we'll have to update this test case, even if the behavior has not changed. If we stick to the props (the public interface/API in React), then we won't have to update this test case.

Reply:
Very good point, thanks for the suggestion.